docs/content/guides/dagster/validating_pandas_dataframes_with_dagster_types.mdx

```diff
@@ -0,0 +1,240 @@
+        "start_time": pa.Column(pd.Timestamp, checks=pa.Check.ge(MIN_DATE)),
+        "end_time": pa.Column(pd.Timestamp, checks=pa.Check.ge(MIN_DATE)),
+    },
+)
+
+schema.validate(df)
+# => SchemaError: non-nullable series 'end_time' contains null values:
+# => 22   NaT
+# => 43   NaT
+```
+
+Our call to `schema.validate` has found the culprit. Somehow, null values have crept into our `end_time` column-- maybe some customers never returned their bikes...
+
+## Dagster Type factory
+
+We are now facing a common situation in data science-- data is frequently dirty. This can cause puzzling bugs if the data travels a significant distance in a pipeline before causing a problem. In our example, it triggered an obtuse numpy error.
+
+We can prevent these situations by validating our data before passing it downstream. There are several ways to do this problem in Dagster. We could inline the schema and validation call in our `load_trips` op. Alternatively, we could break validation into a dedicated op that sits in between `load_trips` and any downstream computations. However, both of these approaches complicate our logic at either the graph or single op level. If this validation strategy were scaled up across all present and future analysis jobs, it's could quickly become a headache. become a headache.
+
+[Dagster Types](/concepts/dagster-types) offer a more robust and elegant solution. Every op input and output can have a Dagster type that encapsulates runtime validation checks. Dagster automatically runs these checks during job execution. So if we can wrap our trip dataframe schema in a Dagster type, then we won't need to further complicate our graphs or individual ops with explicit validation calls and error handling-- Dagster's machinery will take care of this for us.
+
+Fortunately, it's easy to write a factory function that will wrap any Pandera schema in a unique Dagster type:
+
+```python file=/guides/validating_pandas_dataframes_with_dagster_types/factory.py
+from dagster import DagsterType, TypeCheck
+import pandera as pa
+import pandas as pd
+
+def pandera_schema_to_dagster_type(schema, name, description):
+
+    def type_check_fn(_context, value):
+        if not isinstance(value, pd.DataFrame):
+            return TypeCheck(
+                success=False,
+                description=f"Must be pandas.DataFrame, not {type(value).__name__}.",
+            )
+        try:
+            # `lazy` instructs pandera to capture every (not just the first) validation error
+            schema.validate(value, lazy=True)
+        except pa.errors.SchemaErrors as e:
+            return TypeCheck(success=False,
+                description = str(e),
+                metadata = {
+                    'num_violations': len(e.failure_cases),
+                })
+
+        return TypeCheck(success=True)
+
+    return DagsterType(
+      type_check_fn = type_check_fn,
+      name = name,
+      description = description,
```

Comment:
A non-negligible benefit of the current dagster-pandas library is the type description - some users find it valuable to be able to go into Dagit and learn about what columns are expected to be inside an op output.  Would we be able to construct a similarly-informative description using the Pandera schema?